### ISSUE TITLE
Currencies JSON list

### ISSUE THREAD

**Issue by emuhendis** [firefly-iii/firefly-iii#994]
I've just started to use Firefly III.

When creating currencies, wouldn't it be easy if we can choose currencies from a list?

There is a project [https://github.com/xsolla/currency-format](url)

[https://raw.githubusercontent.com/xsolla/currency-format/master/currency-format.json](url)

("JSON with information about currencies: codes (ISO 4217), the names, grapheme (symbols), fraction and formatting.")

It contains all worlds money currencies except sort of coin currencies.

 I'll try to code it. But I am a newbie for Laravel and twig for now.

**Comment by JC5** [comment]
That looks pretty cool!

I could use the list as a source for the import of transaction currencies but unfortunately I cannot use the list directly. I need them to be in the database.

It's also a very long list. I would suggest a sensible selection for each continent, what do you think?

**Comment by emuhendis** [comment]
I think there can be a select list on currency edit/create views. And can be filled by this JSON.

**Comment by JC5** [comment]
Great! I will make sure more currencies will be available in the next release of Firefly III. Not all of them, but more :).

**Comment by emuhendis** [comment]
There is two possibilities for data.

1. Use static JSON list
2. Export JSON list to db.
3. Shrink the list.

My suggestion wasn't that putting all currencies tofirefly currencies page.
On the currencies page, while adding new currency to account, there will be only one search box or select list. No need to other options such as symbol, code etc. Because all needed data will be in JSON or db.

Thanks

**Comment by JC5** [comment]
I get it, thanks for explaining it!

**Comment by JC5** [comment]
Alright, I've been experimenting and I've opted to do the following. Adding currencies using the JSON as a base works but isn't user friendly, with the long list and all. I could add a search of course but I feel it would fix a problem that does not need to exist.

Very few people request new currencies (most people use the built-in ones) and adding them isn't that hard (ie. they don't need that much help).

So, I've decided to expand the list of currencies to include three from each region of the world. This is more or less [an arbitrary pick from my side](https://github.com/firefly-iii/firefly-iii/blob/develop/database/seeds/TransactionCurrencySeeder.php) but I feel that this should cover most people's needs.
